aptos-move/move-examples/data_structures/sources/smart_tree.move

```diff
@@ -0,0 +1,971 @@
+            bucket_table::add(&mut tree.nodes, node_index, node);
+            return
+        };
+
+        if (node_index == tree.root_index) {
+            if (current_size == 1 && !is_leaf) {
+                tree.root_index = vector::borrow(children, 0).node_index;
+                bucket_table::borrow_mut(&mut tree.nodes, tree.root_index).parent = NULL_INDEX;
+            } else {
+                bucket_table::add(&mut tree.nodes, node_index, node);
+            };
+            return
+        };
+
+        // Children size is below threshold.
+
+        let brother_index = next;
+        if (brother_index == NULL_INDEX || bucket_table::borrow(&tree.nodes, brother_index).parent != parent) {
+            brother_index = prev;
+        };
+        let brother_node = bucket_table::remove(&mut tree.nodes, &brother_index);
+        let brother_children = &mut brother_node.children;
+        let brother_size = vector::length(brother_children);
+
+        if ((brother_size - 1) * 2 >= (tree.order as u64)) {
+            // The bother node has enough elements, borrow an element from the brother node.
+            brother_size = brother_size - 1;
+            if (brother_index == next) {
+                let borrowed_element = *vector::borrow(brother_children, 0);
+                vector::push_back(children, borrowed_element);
+                if (borrowed_element.node_index != NULL_INDEX) {
+                    bucket_table::borrow_mut(&mut tree.nodes, borrowed_element.node_index).parent = node_index;
+                };
+                let i = 0;
+                while (i < brother_size) {
+                    *vector::borrow_mut(brother_children, i) = *vector::borrow(brother_children, i + 1);
+                    i = i + 1;
+                };
+                vector::pop_back(brother_children);
+                update_key(tree, parent, vector::borrow(children, current_size - 2).max_key, borrowed_element.max_key);
+            } else {
+                let i = current_size;
+                while (i > 0) {
+                    *vector::borrow_mut(children, i) = *vector::borrow(children, i - 1);
+                    i = i - 1;
+                };
+                let borrowed_element = vector::pop_back(brother_children);
+                if (borrowed_element.node_index != NULL_INDEX) {
+                    bucket_table::borrow_mut(&mut tree.nodes, borrowed_element.node_index).parent = node_index;
+                };
+                *vector::borrow_mut(children, 0) = borrowed_element;
+                update_key(tree, parent, vector::borrow(children, 0).max_key, vector::borrow(brother_children, brother_size - 1).max_key);
+            };
+
+            bucket_table::add(&mut tree.nodes, node_index, node);
+            bucket_table::add(&mut tree.nodes, brother_index, brother_node);
+            return
+        };
+
+        // The bother node doesn't have enough elements to borrow, merge with the brother node.
```

Comment:
@vgao1996  "The bother" -> "The brother"

Reply:
Done